isodesign/process/calculation.py

```diff
@@ -1,113 +1,94 @@
-        """
         self.num_carbon = len(self.labeling)
+        self.step = step
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+        self.fraction = [fraction for fraction in range(self.lower_bound, self.upper_bound + self.step, self.step)]
+       
+    def __len__(self):
+        return self.num_carbon
 
     def __repr__(self) -> str:
         return f"Molecule name: {self.name}, Number of associated carbon(s) : {self.num_carbon}"
-
-
-def intra_mol_comb(num_mol, step=10, start=0, end=100):
-    """
-    Building the intramolecular cominations. Default values :
-        * start: 0
-        * end: 100
-
-    :param num_mol: number of molecules into the family 
-    :param step: 
     
-    :return: list of tuple containing all the intramolecular combinations
-    """
-
-    all_fracs = [] #list containing list of fractions in function of the number of molecule 
-    for _ in range(len(num_mol)):
-        fractions=[data for data in range(start,end,step)] #list of fractions 
-        all_fracs.append(fractions) #adding list of fraction for a molecule in the fraction list 
-    intra_combi = [pair for pair in list(product(*all_fracs)) if sum(pair) == 100] #list containing all the combination inside a molecule family  
-    return intra_combi
-
-
-def inter_mol_comb(fam_mol1,fam_mol2, step=10, start=0, end=100):
-    """ 
-    Building the intermolecular combinations. Default values :
-        * start: 0
-        * end: 100 
+    @property
+    def lower_bound(self):
+        return self._lower_bound
     
-    :param fam_mol1: list of molecule names of the first molecule family   
-    :param fam_mol2: list of molecule names of the second molecule family
-    :param step: 
-
-    :return: list containing tuple which contain tuple with all the intermolecular combinations
-
-    Using the function intra_mol_comb for each molecule family to have the 
-    list of combination inside the molecule family. 
-    """
+    @lower_bound.setter
+    def lower_bound(self, value):
+        if value < 0:
+            raise ValueError(f"{value} isn't correct. Use a positive value.")
+        self._lower_bound = value
+
+    @property
+    def upper_bound(self):
+        return self._upper_bound
     
-    combi_family_1 = intra_mol_comb(fam_mol1) #intramolecular combination for the first family 
-    combi_family_2 = intra_mol_comb(fam_mol2) #intramolecular combination for the second family
-    intermol_combi = list(product(combi_family_1,combi_family_2)) #list containing all the combination
-    return intermol_combi
-
-def generate_file(fam_mol1, fam_mol2, label_fam1, label_fam2):
-    """ 
-    Generating .linp files in function of all combination between 2 molecule family. 
+    @upper_bound.setter
+    def upper_bound(self, value):
+        if value > 100:
+            raise ValueError(f"{value} isn't correct. Value must be less than 100.")
+        self._upper_bound = value
```

Comment:
Même chose ici. Je mettrais plutôt "proportions are given as a percentage and thus cannot be higher than 100%". Information que l'on devrait mettre dans les docstrings aussi.
Et y'a aussi au moins une autre condition que je vois pour les upper bounds